### PR TITLE
#13 implemented a way to display play errors on the website

### PR DESCRIPTION
**Pull request by SailReal**
see #13

### CODE REVIEW

File: app/typescript/model/GameState.ts [hunk 1]
```diff
@@ -4,4 +4,5 @@ interface GameState {
     currentQuestion?: Question;
     currentQuestionTime?: number;
     helpText: string;
+    errorMessage: string;
```
Comment (DonatJR) [inline app/typescript/model/GameState.ts]: this property should be optional, just add a `?` to the name

File: app/controllers/RestController.scala [hunk 2]
```diff
@@ -73,7 +75,26 @@ class RestController @Inject()(
...
+    def execute(fn: => Unit): Unit = {
+        try {
+            fn
+        } catch {
+            case e: ControllerException =>
+                sendJsonToWebSocket(createErrorJson(e.message))
+        }
+    }
+
+    def sendJsonToWebSocket(json: JsValue): Unit = {
```
Comment (DonatJR) [inline app/controllers/RestController.scala]: is it necessary to have this in the `HomeController` and `RestController`? if we need it here we could just make the def in the `HomeController` public

File: app/controllers/RestController.scala [hunk 3]
```diff
@@ -14,55 +15,56 @@ import scala.concurrent.Future
 class RestController @Inject()(
     cc: ControllerComponents,
     silhouette: Silhouette[DefaultEnv],
-    serverCtrl: Controller
+    serverCtrl: Controller,
+    homectrl: HomeController
```
Comment (DonatJR) [inline app/controllers/RestController.scala:19]: I kind of dislike sharing controller instances within each other :/ would be awesome if we could somehow just share a single method.

Thoughts for the future: Maybe the actor-management should be its own class with then could be shared with home and rest controller

Comment (SailReal) [reply app/controllers/RestController.scala:19]: You're right, I had the same feeling as it came to inject another controller. 
Putting the actor-management in an own class sounds nice, I will create an issue for that but would say it's an 1.x "feature"?

Comment (DonatJR) [reply app/controllers/RestController.scala:19]: yep, definitely 1.x

**Review by DonatJR — APPROVED** [review]

**Pull request merged**